### ISSUE TITLE
JSON value {...} of type NSMUtableDictionary cannot be converted to NSArray

### ISSUE THREAD

**Issue by marcojr** [dooboolab/react-native-iap#387]
![simulator screen shot - iphone x - 2019-01-29 at 11 01 04](https://user-images.githubusercontent.com/856617/51903901-395d9d80-23b5-11e9-9e51-1f3e82e9e427.png)
### Steps to reproduce the behavior

react-native init myApp

cd myApp

npm install react-native-iap --save

react-native link react-native-iap

App.js
import * as RNIap from 'react-native-iap';
async componentDidMount() {
        const itemSkus = {
            ios: [
                'pic1',
            ],
            android: [
                'point_1000',
            ]
        };
        try {
            const products = await RNIap.getProducts(itemSkus);
            this.setState({ products });
        } catch(err) {
            console.warn(err); // standardized err.code and err.message available
        }
    }

Basically I created a new fresh app and used the very examples on the git page - And nothing else.

**Comment by hyochan** [comment]
@marcojr I think I need to dig this more to see which actually causes the problem. `react-native@0.58`?

**Comment by marcojr** [comment]
> @marcojr I think I need to test this out to see which actually causes the problem. `react-native@0.58`?

Yes, sir ! 0.58.3

As I mentioned before: Just start a new react-native app, follow your instructions to install (automatic, react-native-link), use your own example of the git page... this is all I did.

**Comment by hyochan** [comment]
@marcojr Yeah. I have to test if this is `react-native@0.58` problem or our module by installing them and run it. I'll comeback to you after that.

**Comment by marcojr** [comment]
Thanx @hyochan !

**Comment by marcojr** [comment]
More possible pertinent details to help:

Xcode: 10.1
iOS: 12.1 (Emulator)

**Comment by hyochan** [comment]
@marcojr Thanks. I also hope you to try out last release version which is `2.3.23` to scope down the problem area.

**Comment by neilco** [comment]
@marcojr The problem here is how you’re defining your `itemSkus`:

```javascript
const itemSkus = {
  ios: [
    'pic1',
  ],
  android: [
    'point_1000',
  ]
};
```

You’re missing `Platform.select`. That is:

```javascript
const itemSkus = Platform.select({
  ios: [
    'pic1',
  ],
  android: [
    'point_1000',
  ]
});
```

**Comment by hyochan** [comment]
Oh you are right. I can't believe how I've missed this.
